R/shared-methods.R

```diff
@@ -509,6 +513,65 @@ DataClass <- R6::R6Class(
         source_data_cols = paste(unlist(self$source_data_cols), collapse = ", ")
       )
       return(sum_df)
+    },
+
+    #' @description Run tests on a country class instance. Calling `test()` on a
+    #' class instance runs tests with the settings in use. For example, if you
+    #' set `level = "1"` and `localise = FALSE` the tests will be run on level 1
+    #' data which is not localised. Rather than downloading data for a test
+    #' users can provide a path to a snapshot file of data to test instead.
+    #' Tests are run on a clone of the class. This method calls generic tests
+    #' for all country class objects. It also calls country specific tests
+    #' which can be defined in an individual country class method called
+    #' `specific_tests()`. The snapshots contain the first 1000 rows of data.
+    #' For more details see the 'testing' vignette.
+    #' @param download logical. To download the data (TRUE) or use a snapshot
+    #' (FALSE). Defaults to FALSE.
+    #' @param snapshot_dir character_array the name of a directory to save the
+    #' downloaded data or read from. If not defined a directory called
+    #' 'snapshots' will be created in the current working directory. Snapshots
```

Comment:
we can't default to saving in the current directory as CRAN will get mad. We need to have the default be tempdir().